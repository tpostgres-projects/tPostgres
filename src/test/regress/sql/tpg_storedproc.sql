--
-- Stored Procedures
--

--
-- Test BEGIN and END as subset of word tokens as well as being part of string
-- literals.
--

create procedure begin_blockdepth_check as
declare beginning varchar(20) = 'begin'
begin
    print beginning
end

create procedure end_blockdepth_check as
declare ending varchar(20) = 'end'
begin
    print ending
end

select begin_blockdepth_check();
select end_blockdepth_check();

-- Create a couple of overloaded procedures
create procedure overloaded_proc(@a text) as
begin
    print @a
end

create procedure overloaded_proc(@a int) as
begin
    print @a
end

--
-- Test DROP PROCEDURE
--

-- Ambiguous: should fail
drop procedure overloaded_proc;
-- Should succeed.
drop function overloaded_proc(text);
-- This should succeed too.
drop procedure overloaded_proc;
-- no-op without an error
drop procedure if exists overloaded_proc;