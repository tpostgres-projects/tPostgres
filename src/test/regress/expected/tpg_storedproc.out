--
-- Stored Procedures
--
--
-- Test BEGIN and END as subset of word tokens as well as being part of string
-- literals.
--
create procedure begin_blockdepth_check as
declare beginning varchar(20) = 'begin'
begin
    print beginning
end
create procedure end_blockdepth_check as
declare ending varchar(20) = 'end'
begin
    print ending
end
select begin_blockdepth_check();
INFO:  begin
 begin_blockdepth_check 
------------------------
 
(1 row)

select end_blockdepth_check();
INFO:  end
 end_blockdepth_check 
----------------------
 
(1 row)

-- Create a couple of overloaded procedures
create procedure overloaded_proc(@a text) as
begin
    print @a
end
create procedure overloaded_proc(@a int) as
begin
    print @a
end
--
-- Test DROP PROCEDURE
--
-- Ambiguous: should fail
drop procedure overloaded_proc;
ERROR:  function overloaded_proc() is not unique
HINT:  You might need to use DROP FUNCTION with an explicit list of parameter types.
-- Should succeed.
drop function overloaded_proc(text);
-- This should succeed too.
drop procedure overloaded_proc;
-- no-op without an error
drop procedure if exists overloaded_proc;
NOTICE:  function overloaded_proc() does not exist, skipping
